File: docs/snippets/all/archetypes/cylinders3d_batch.py
"""Log a batch of cylinders."""

import rerun as rr

rr.init("rerun_example_cylinders3d_batch", spawn=True)

rr.log(
    "cylinders",
    rr.Cylinders3D(
        lengths=[0.0, 2.0, 4.0, 6.0, 8.0],
        radii=[1.0, 0.5, 0.5, 0.5, 1.0],
        colors=[
            (255, 0, 0),
            (188, 188, 0),
            (0, 255, 0),
            (0, 188, 188),
            (0, 0, 255),
        ],
        centers=[
            (0.0, 0.0, 0.0),
            (2.0, 0.0, 0.0),
            (4.0, 0.0, 0.0),
            (6.0, 0.0, 0.0),
            (8.0, 0.0, 0.0),
        ],
        rotation_axis_angles=[
            rr.RotationAxisAngle(
                [1.0, 0.0, 0.0],
                rr.Angle(deg=float(i) * -22.5),
            )
            for i in range(5)
        ],
    ),
)

File: docs/snippets/all/archetypes/cylinders3d_batch.cpp
// Log a batch of cylinders.

#include <rerun.hpp>

int main(int argc, char* argv[]) {
    const auto rec = rerun::RecordingStream("rerun_example_cylinders3d_batch");
    rec.spawn().exit_on_failure();

    rec.log(
        "cylinders",
        rerun::Cylinders3D::from_lengths_and_radii(
            {0.0f, 2.0f, 4.0f, 6.0f, 8.0f},
            {1.0f, 0.5f, 0.5f, 0.5f, 1.0f}
        )
            .with_colors({
                rerun::Rgba32(255, 0, 0),
                rerun::Rgba32(188, 188, 0),
                rerun::Rgba32(0, 255, 0),
                rerun::Rgba32(0, 188, 188),
                rerun::Rgba32(0, 0, 255),
            })
            .with_centers({
                {0.0f, 0.0f, 0.0f},
                {2.0f, 0.0f, 0.0f},
                {4.0f, 0.0f, 0.0f},
                {6.0f, 0.0f, 0.0f},
                {8.0f, 0.0f, 0.0f},
            })
            .with_rotation_axis_angles({
                rerun::RotationAxisAngle({1.0f, 0.0f, 0.0f}, rerun::Angle::degrees(0.0)),
                rerun::RotationAxisAngle({1.0f, 0.0f, 0.0f}, rerun::Angle::degrees(-22.5)),
                rerun::RotationAxisAngle({1.0f, 0.0f, 0.0f}, rerun::Angle::degrees(-45.0)),
                rerun::RotationAxisAngle({1.0f, 0.0f, 0.0f}, rerun::Angle::degrees(-67.5)),
                rerun::RotationAxisAngle({1.0f, 0.0f, 0.0f}, rerun::Angle::degrees(-90.0)),
            })
    );
}
